lib/User/Manager.php

```diff
@@ -240,6 +240,18 @@ public function getCachedEntry($uid) {
 			return $this->usersByUid[$uid];
 		}
 
+		try {
+			$dn = $this->username2dn($uid);
+			if ($dn) {
+				return $this->getUserEntryByDn($dn);
+			}
+		} catch (DoesNotExistOnLDAPException $e) {
+			return null;
+		} catch (\OutOfBoundsException $e) {
+			// dn might be outside of the base
+			return null;
+		}
```

Comment:
Was this required to restore the avatar behaviour? It should just be the below line that is required. The reason I am concerned with this change is that when we optimised the sync code, it should not need to do single entry retrievals from LDAP, they should be cached when loading pages of results. Hence we did not need to load them manually here. What was the reason for the change here @jvillafanez ?

Reply:
Yes, it's required. I don't remember the exact use case, but I'm pretty sure that without this change it crashes because the entry isn't cached yet.
Maybe https://github.com/owncloud/core/blob/5a9909dde20c753c6a4c018e8dba83012ed08756/settings/Panels/Personal/Profile.php#L101 . You have to check the backend there in order to check if the user has an avatar set or not in the LDAP server.

Reply:
Maybe it doesn't crashes, but it doesn't behave according to the expectations. Quoting the original ticket:
> A user without avatar in LDAP can add his own avatar.
A user with an avatar in LDAP cannot set his own avatar.

If the user isn't cached as it seems, the LDAP user won't be able to set his avatar in any circumstance.